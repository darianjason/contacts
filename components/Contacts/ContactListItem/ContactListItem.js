import React from 'react';
import {View, Pressable} from 'react-native';

import {Card, DefaultText, ProfilePicture} from '../../ui';
import styles from './ContactListItem.styles';

const NameContainer = ({firstName, lastName}) => (
  <View style={styles.nameContainer}>
    <DefaultText style={styles.name}>
      {firstName} {lastName}
    </DefaultText>
  </View>
);

const ContactListItem = ({firstName, lastName, age, photo, onSelect}) => (
  <Pressable
    onPress={onSelect}
    style={({pressed}) => pressed && styles.pressed}
    accessibilityLabel="Contact list item"
  >
    <Card style={styles.itemContainer}>
      <ProfilePicture photo={photo} firstName={firstName} lastName={lastName} />
      <NameContainer firstName={firstName} lastName={lastName} />
    </Card>
  </Pressable>
);

export default ContactListItem;
