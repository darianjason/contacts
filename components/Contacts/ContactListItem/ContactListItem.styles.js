import {StyleSheet, Dimensions} from 'react-native';

import {Colors} from '../../../constants';

const styles = StyleSheet.create({
  pressed: {
    opacity: 0.5,
  },
  itemContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginVertical: 8,
  },
  imageContainer: {
    width: Dimensions.get('window').height / 12,
    height: Dimensions.get('window').height / 12,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.accent,
    borderRadius: 8,
    overflow: 'hidden',
  },
  image: {
    width: '100%',
    height: '100%',
  },
  initials: {
    fontSize: 16,
    fontFamily: 'Manrope-Bold',
    letterSpacing: 2,
    color: Colors.primary,
  },
  nameContainer: {
    padding: 16,
  },
});

export default styles;
