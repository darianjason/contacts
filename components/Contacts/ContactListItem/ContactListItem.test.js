import React from 'react';
import {render} from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import ContactListItem from './ContactListItem';

describe('ContactListItem component', () => {
  test('renders full name', () => {
    const firstName = 'John';
    const lastName = 'Doe';

    const {getByText} = render(
      <ContactListItem firstName={firstName} lastName={lastName} />,
    );

    expect(getByText('John Doe')).toBeTruthy();
  });

  test('renders correctly', () => {
    const firstName = 'John';
    const lastName = 'Doe';
    const age = 25;
    const photo = 'uri';
    const onSelect = jest.fn();

    const tree = renderer
      .create(
        <ContactListItem
          firstName={firstName}
          lastName={lastName}
          age={age}
          photo={photo}
          onSelect={onSelect}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
