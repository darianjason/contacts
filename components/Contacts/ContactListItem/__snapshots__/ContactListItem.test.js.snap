// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactListItem component renders correctly 1`] = `
<View
  accessibilityLabel="Contact list item"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginVertical": 8,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#E2EEF3",
          "borderRadius": 8,
          "height": 111.16666666666667,
          "justifyContent": "center",
          "overflow": "hidden",
          "width": 111.16666666666667,
        }
      }
    >
      <Image
        accessibilityRole="image"
        source={
          Object {
            "uri": "uri",
          }
        }
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "padding": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#060A0E",
            "fontFamily": "Manrope-Regular",
          }
        }
      >
        John
         
        Doe
      </Text>
    </View>
  </View>
</View>
`;
